### ISSUE TITLE
Invalid OCAD 10 export

### ISSUE THREAD

**Issue by jmacura** [OpenOrienteering/mapper#1234]
### Steps to reproduce
1. Create map in OCAD 10
2. Open the .ocd map in OO Mapper
3. Save the map as .omap file
4. Save the map as OCAD 10 format again (new file, do not rewrite)
5. Open the file in OCAD 10

### Actual behaviour
OCAD 10 complains with "Internal error: BackgroundMapOptions.LoadBackgroundMap" 2 times. The rest is a bit strange – the first time I've tried it, the loading process has finished, but the map objects were invisible (selectable but not visible). The second, third, fourth time, the message "Access violation at address 00403BCB in module 'Ocad10Pro.exe'. Write of address 00000000." appeared and the load process won't finish (half-stuck program, neither freezed, GUI responding, but not doing any operation).

### Expected behaviour
OCAD should open the file like it was its own.

### Configuration
Mapper Version: unstable 0.9.20190326
Operating System: Windows 10 (v. 1803)

**Comment by lpechacek** [comment]
> "Access violation at address 00403BCB in module 'Ocad10Pro.exe'. Write of address 00000000."

Cool. A possibly exploitable bug in OCAD. If you want to be nice to OCAD guys, bring the error message to their attention.

Reading the description, I can see the files are empty and perhaps shareable. Would you be so kind as to attach _both_ the files to this report, please?

**Comment by lpechacek** [comment]
And thanks for testing the development version!

**Comment by jmacura** [comment]
I sincerely doubt that OCAD guys care about the OCAD 10 version any more... ;-)
I would like to attach the files but don't like the idea of publicly presenting a (although unfinished) map for an upcoming event. I can either a) send them to you privately or b) attach the files here when the event is over (mid June).
Thanks!

**Comment by lpechacek** [comment]
I see. I misunderstood the description. "Create map" stands for creating the whole map work, not an empty file. I can see OCAD complains about background images. To further minimize the amount of data shared, would you create a copy of your map file with all objects removed (empty map, just symbols, colors and the background images) and retry your test? You can continue with removing colors and individual templates until arriving at the smallest file triggering the crash. Always keep a copy of the file from previous step as a minimum. Thanks!

**Comment by jmacura** [comment]
Exactly, it is a map with quite a few objects already, definitely not empty map :-)
Thanks for the tip, I will play with file conversion later this week when I found some time.

**Comment by dg0yt** [comment]
We should have an eye on the encoding of the strings defining the templates. Version 9 used some Windows 8-bit encoding, while version 11 used UTF-8. I wouldn't be much surprised with strange results for non-latin characters in the version between 9 and 11. 
OTOH, the BackgroundMap thing might be a red herring, appearing long after where things started to go wrong.

**Comment by jmacura** [comment]
@lpechacek So I've elaborated a bit this evening.
"Sadly", _I can not reproduce the error again at all._ I've tried to import the OCAD 10 map into Mapper again, save the file as .omap, save the file as OCAD 10 and it just works. OCAD still complains about the InternalError of templates (which is acceptable as some of the background files are really missing), but finishes loading and displays the map, as expected.
So I've checked, in which way the files created today and week ago are different, and found out that the .omap files differ in their content (some objects missing, some added). And sadly, by checking the date of the last change of the files, it seems as I've made some additional edits in the original OCAD map, and that I do not posses the exact same original .ocd file with which I hit this weird error last week...
I can not say why these edits made the behaviour of import/export in Mapper different, but they probably did.
I propose to close this issue as unreproducible. If I eventually encounter it again, I will re-open it.
